Sources/StreamChat/Workers/TypingEventsSender.swift

```diff
@@ -13,34 +13,40 @@ extension TimeInterval {
     static let startTypingResendInterval: TimeInterval = .incomingTypingStartEventTimeout - .startTypingEventTimeout
 }
 
+/// Struct for storing currently typing info.
+private struct TypingInfo {
+    let channelId: ChannelId
+    let parentMessageId: MessageId?
+}
+
 /// Sends typing events.
 class TypingEventsSender: Worker {
     /// A timer type.
     var timer: Timer.Type = DefaultTimer.self
-    /// ChannelId for channel that typing has occurred in. Stored to stop typing when `TypingEventsSender` is deallocated
-    private var typingChannelId: ChannelId?
+    /// `TypingInfo` for channel (and parent message) that typing has occurred in. Stored to stop typing when `TypingEventsSender` is deallocated
+    private var typingInfo: TypingInfo?
```

Comment:
Would it work fine when the user quickly switches threads?

Reply:
It's exactly the same logic we had before, so if it was working before...